src/Test/WorkflowApplicationTestExtensions/Persistence/DataContractWorkflowSerializer.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System.IO;
+using System.Activities.Runtime.DurableInstancing;
+using System.Collections.Generic;
+using System.Xml.Linq;
+using System.Runtime.Serialization;
+using System.Collections;
+using System.Xml;
+using System;
+using System.Activities;
+using System.Reflection;
+
+namespace WorkflowApplicationTestExtensions.Persistence;
+
+
+using XInstanceDictionary = IDictionary<XName, InstanceValue>;
+using InstanceDictionary = Dictionary<string, InstanceValue>;
+
+public interface IFaultHandler
+{
+    void OnFault(NativeActivityFaultContext faultContext, Exception propagatedException, ActivityInstance propagatedFrom);
+    void OnComplete(NativeActivityContext context, ActivityInstance completedInstance);
+}
+
+public sealed class DataContractWorkflowSerializer : IWorkflowSerializer
+{
+    public XInstanceDictionary LoadWorkflowInstance(Stream sourceStream) =>
+            WorkflowSerializerHelpers.ToNameDictionary(GetDataContractSerializer().ReadObject(sourceStream));
+
+    public void SaveWorkflowInstance(XInstanceDictionary workflowInstanceState, Stream destinationStream) =>
+        GetDataContractSerializer().WriteObject(destinationStream, workflowInstanceState.ToSave());
+
+    private static DataContractSerializer GetDataContractSerializer()
+    {
+        DataContractSerializerSettings settings = new DataContractSerializerSettings
+        {
+            PreserveObjectReferences = true,
+            DataContractResolver = new WorkflowDataContractResolver()
+        };
+        var dataContractSerializer = new DataContractSerializer(typeof(InstanceDictionary), settings);
+        dataContractSerializer.SetSerializationSurrogateProvider(new SerializationSurrogateProvider());
```

Comment:
The whole surrogate thing should be removed.

Reply:
here, I was thinking maybe we think of something taht is actually included with CoreWf, ...or whatever that can be shared tests/production